x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/action/TransportDeleteDataFrameAnalyticsAction.java

```diff
@@ -242,17 +241,46 @@ private void deleteState(ParentTaskAssigningClient parentTaskClient,
                              DataFrameAnalyticsConfig config,
                              TimeValue timeout,
                              ActionListener<BulkByScrollResponse> listener) {
-        List<String> ids = new ArrayList<>();
-        ids.add(StoredProgress.documentId(config.getId()));
-        if (config.getAnalysis().persistsState()) {
-            ids.add(config.getAnalysis().getStateDocId(config.getId()));
+        ActionListener<Boolean> deleteModelStateListener = ActionListener.wrap(
```

Comment:
Does `UnusedStateRemover` class need to be changed too?

Reply:
No because the corresponding job id extractors (see `Classification.extractJobIdFromStateDoc` and `Regression.extractJobIdFromStateDoc`) successfully extract the job id regardless of the document number that follows the infix.